### PR TITLE
feat(addons-react): create withFluxible and useFluxible utilities

### PR DESCRIPTION
**Pull request by pablopalacios**
These were the missing pieces to provide fluxible context to components without relying on `contextType`.

Usage:

```javascript
import { withFluxible, useFluxible } from 'fluxible-addons-react';

// Functional components
const Component = () => {
    const { executeAction, getStore } = useFluxible();
    return null;
} 

// Class components
class Component extends React.Component {
    render() {
       const { executeAction, getStore } = this.props.context;
       return null;
    }
}

export default withFluxible(Component); 
```
I also renamed `FluxibleComponent` to `FluxibleComponentContext` so it doesn't clash with `fluxible.FluxibleContext` class.
 

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by redonkulus — COMMENTED** [review]
Do you want to update examples and docs in this PR as well?

**Comment by pablopalacios** [comment]
I will do examples in another one since it will probably trigger way more changes. But I'll send the docs.

**Comment by pablopalacios** [comment]
@redonkulus I added the documentation entries for functions I created. While doing this, I also realized that the overall docs must be completed updated (I missed that in someway). I'll create another PR for the whole docs update.

File: packages/fluxible-addons-react/docs/api/withFluxible.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  render() {
+    const { context } = this.props;
+
+    return (
+      <button onClick={() => context.executeAction(myAction)} />
+    );
+  }
+}
+
+export default withContext(Component);
```
Comment (redonkulus) [inline packages/fluxible-addons-react/docs/api/withFluxible.md:23]: Should this be withFluxible?

**Review by redonkulus — COMMENTED** [review]
Looks good, just one question

File: packages/fluxible-addons-react/docs/api/withFluxible.md (hunk 1 above)
Comment (pablopalacios) [reply packages/fluxible-addons-react/docs/api/withFluxible.md]: Fixed.

**Review by redonkulus — APPROVED** [review]
Thanks again. Changes look good. Let’s make sure the upgrade guide and site docs are updated to reflect these changes.

**Pull request merged**